apps/aecore/test/aecore_txs_SUITE.erl

```diff
@@ -0,0 +1,141 @@
+-module(aecore_txs_SUITE).
+
+%% This code is brutaly copied form aecore_sync_SUITE and should use joined code base.
+
+%% common_test exports
+-export(
+   [
+    all/0,
+    init_per_suite/1, end_per_suite/1,
+    init_per_testcase/2, end_per_testcase/2
+   ]).
+
+%% test case exports
+-export(
+   [ txs_gc/1
+   ]).
+
+
+-include_lib("common_test/include/ct.hrl").
+
+all() ->
+    [ txs_gc
+    ].
+
+init_per_suite(Config) ->
+    ok = application:ensure_started(erlexec),
+    DataDir = ?config(data_dir, Config),
+    TopDir = aecore_suite_utils:top_dir(DataDir),
+    Config1 = [{symlink_name, "latest.txs"},
+               {top_dir, TopDir},
+               {test_module, ?MODULE}] ++ Config,
+    aecore_suite_utils:make_shortcut(Config1),
+    ct:log("Environment = ~p", [[{args, init:get_arguments()},
+                                 {node, node()},
+                                 {cookie, erlang:get_cookie()}]]),
+    aecore_suite_utils:create_configs(Config1),
+    aecore_suite_utils:make_multi(Config1),
+    [{nodes, [aecore_suite_utils:node_tuple(dev1)]} | Config1].
+
+end_per_suite(Config) ->
+    aecore_suite_utils:stop_node(dev1, Config),
+    ok.
+
+init_per_testcase(_Case, Config) ->
+    ct:log("testcase pid: ~p", [self()]),
+    [{tc_start, os:timestamp()}|Config].
+
+end_per_testcase(_Case, Config) ->
+    Ts0 = ?config(tc_start, Config),
+    ct:log("Events during TC: ~p", [[{N, aecore_suite_utils:all_events_since(N, Ts0)}
+                                     || {_,N} <- ?config(nodes, Config)]]),
+    ok.
+
+%% ============================================================
+%% Test cases
+%% ============================================================
+txs_gc(Config) ->
+    aecore_suite_utils:start_node(dev1, Config),
+    N1 = aecore_suite_utils:node_name(dev1),
+    aecore_suite_utils:connect(N1),
+
+    %% Mine a block to get some funds.
+    aecore_suite_utils:mine_blocks(N1, 1),
+
+    %% Add a bunch of transactions...
+    add_spend_tx(N1, 1000, 1,  1,  10), %% Ok
+    add_spend_tx(N1, 1000, 1,  2,  10), %% Should expire ?EXPIRE_TX_TTL after
+                                        %% next TX is on chain = 2 + 2 = 4
+    add_spend_tx(N1, 1000, 10, 2,  10), %% Duplicate should be preferred
+    add_spend_tx(N1, 1000, 1,  3,  10), %% Ok
+
+    add_spend_tx(N1, 1000, 1,  5,  10), %% Non consecutive nonce
+    add_spend_tx(N1, 1000, 1,  7,  10), %% Non consecutive nonce
+    add_spend_tx(N1, 1000, 1,  8,  4),  %% Short TTL - expires at 4 + 2 = 6
```

Comment:
I see that the specified 4 is the TTL as absolute height, and the magic 2 comes from [here](https://github.com/aeternity/epoch/pull/1255/commits/249b6a3e538029cc020f8f07f1b2a7f976933a5d?utf8=%E2%9C%93&diff=split&w=1#diff-2e95a69d7723aebad303814cb3f38e7eR80). Ok.